src/Plugin/EntityReferenceSelection/SkosConceptSelection.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\rdf_skos\Plugin\EntityReferenceSelection;
+
+use Drupal\Core\Entity\Plugin\EntityReferenceSelection\DefaultSelection;
+use Drupal\Core\Entity\Query\QueryInterface;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Selection plugin for SKOS Concepts.
+ *
+ * @EntityReferenceSelection(
+ *   id = "default:skos_concept",
+ *   label = @Translation("SKOS Concept selection"),
+ *   entity_types = {"skos_concept"},
+ *   group = "default",
+ *   weight = 1
+ * )
+ */
+class SkosConceptSelection extends DefaultSelection {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration(): array {
+    return [
+      // Empty array means allow all.
+      'concept_schemes' => [],
+    ] + parent::defaultConfiguration();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state): array {
+    $form = parent::buildConfigurationForm($form, $form_state);
+
+    $configuration = $this->getConfiguration();
+    $concept_scheme_ids = $this->entityManager->getStorage('skos_concept_scheme')->getQuery()
+      ->execute();
+
+    $form['concept_schemes'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Concept Schemes'),
+      '#description' => $this->t('Concept Schemes to filter by. Leave empty to allow all.'),
+      '#options' => [],
+      '#default_value' => array_values($configuration['concept_schemes']),
+      '#multiple' => TRUE,
+    ];
+
+    if (!$concept_scheme_ids) {
+      return $form;
+    }
+
+    /** @var \Drupal\rdf_skos\Entity\ConceptSchemeInterface[] $concept_schemes */
+    $concept_schemes = $this->entityManager->getStorage('skos_concept_scheme')->loadMultiple($concept_scheme_ids);
+    $options = [];
+    foreach ($concept_schemes as $concept_scheme) {
+      $options[$concept_scheme->id()] = $concept_scheme->getTitle();
+    }
+
+    $form['concept_schemes'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Concept Schemes'),
+      '#description' => $this->t('Concept Schemes to filter by. Leave empty to allow all.'),
+      '#options' => $options,
+      '#default_value' => array_values($configuration['concept_schemes']),
+      '#multiple' => TRUE,
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateConfigurationForm(array &$form, FormStateInterface $form_state): void {
+    parent::validateConfigurationForm($form, $form_state);
+
+    $settings = $form_state->getValue('settings');
+    $concept_schemes = $settings['handler_settings']['concept_schemes'];
+    if ($concept_schemes) {
```

Comment:
Shouldn't we use more strict and readable comparison constructions?

Reply:
Sure.

Reply:
Nothing wrong with this.

Reply:
It's linked to my other comment, I think we should make the conditions more verbose... using the exclamation mark can be really weird sometimes, see: https://3v4l.org/rTLlU